src/control/server/storage/bdev_test.go

```diff
@@ -0,0 +1,211 @@
+//
+// (C) Copyright 2018-2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package storage
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/daos-stack/daos/src/control/logging"
+)
+
+// TestParseBdev verifies config parameters for bdev get converted into nvme
+// config files that can be consumed by spdk.
+func TestParseBdev(t *testing.T) {
+	tests := map[string]struct {
+		bdevClass  BdevClass
+		bdevList   []string
+		bdevSize   int // relevant for MALLOC/FILE
+		bdevNumber int // relevant for MALLOC
+		vosEnv     string
+		wantBuf    []string
+		errMsg     string
+	}{
+		"defaults from example config": {
+			wantBuf: []string{
+				`[Nvme]`,
+				`    TransportID "trtype:PCIe traddr:0000:81:00.0" Nvme__0`,
+				`    RetryCount 4`,
+				`    TimeoutUsec 0`,
+				`    ActionOnTimeout None`,
+				`    AdminPollRate 100000`,
+				`    HotplugEnable No`,
+				`    HotplugPollRate 0`,
+				``,
+			},
+		},
+		"multiple controllers": {
+			bdevClass: BdevClassNvme,
+			bdevList:  []string{"0000:81:00.0", "0000:81:00.1"},
+			wantBuf: []string{
+				`[Nvme]`,
+				`    TransportID "trtype:PCIe traddr:0000:81:00.0" Nvme__0`,
+				`    TransportID "trtype:PCIe traddr:0000:81:00.1" Nvme__1`,
+				`    RetryCount 4`,
+				`    TimeoutUsec 0`,
+				`    ActionOnTimeout None`,
+				`    AdminPollRate 100000`,
+				`    HotplugEnable No`,
+				`    HotplugPollRate 0`,
+				``,
+			},
+		},
+		"AIO file": {
+			bdevClass: BdevClassFile,
+			bdevList:  []string{"myfile", "myotherfile"},
+			bdevSize:  5, // GB/file
+			wantBuf: []string{
+				"[AIO]",
+				"    AIO myfile AIO__0 4096",
+				"    AIO myotherfile AIO__1 4096",
+				"",
+			},
+			vosEnv: "AIO",
+		},
+		"AIO kdev": {
+			bdevClass: BdevClassKdev,
+			bdevList:  []string{"sdb", "sdc"},
+			wantBuf: []string{
+				"[AIO]",
+				`    AIO sdb AIO__0`,
```

Comment:
is the use of backticks on some lines and quotes on others intentional?

Reply:
that's how it was laid out before, will fix to be consistent in a follow-up patch